[Bug] Kick not functioning / TAB Complete Issues

### Description

While typing the /f kick <player> command, the console repeatedly sends the same red TAB Completion ERROR after each new letter written.

When the command is sent, the player is not kicked. Instead, a WARN message appears about using outdated way of handling exceptions, with no further errors.

The attached log includes only the final red error, as it is identical each time.

### Steps To Reproduce

1. Run the "/f kick <name>" command.

### Log

[23:45:21 ERROR]: Exception when n0ttws attempted to tab complete kick WardenSky
org.bukkit.command.CommandException: Unhandled exception during tab completion for command '/f kick WardenSky' in plugin ImprovedFactions v2.3.1
        at org.bukkit.command.PluginCommand.tabComplete(PluginCommand.java:150) ~[paper-api-1.21.8-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.command.Command.tabComplete(Command.java:100) ~[paper-api-1.21.8-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigSuggestionProvider.getSuggestions(BukkitCommandNode.java:109) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at com.mojang.brigadier.tree.ArgumentCommandNode.listSuggestions(ArgumentCommandNode.java:71) ~[brigadier-1.3.10.jar:1.21.8-60-29c8822]
        at com.mojang.brigadier.CommandDispatcher.getCompletionSuggestions(CommandDispatcher.java:550) ~[paper-1.21.8.jar:?]
        at com.mojang.brigadier.CommandDispatcher.getCompletionSuggestions(CommandDispatcher.java:530) ~[paper-1.21.8.jar:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.sendServerSuggestions(ServerGamePacketListenerImpl.java:904) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleCustomCommandSuggestions0$2(ServerGamePacketListenerImpl.java:863) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:155) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1450) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:176) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:129) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1430) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1424) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:139) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:1381) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1389) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1266) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:310) ~[paper-1.21.8.jar:1.21.8-60-29c8822]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NullPointerException
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.handleTabCompletion(KickCommandProcessor.kt:70) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.access$handleTabCompletion(KickCommandProcessor.kt:21) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor$tabComplete$$inlined$loggedTransaction$1.invoke(DatabaseManager.kt:30) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor$tabComplete$$inlined$loggedTransaction$1.invoke(DatabaseManager.kt:23) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:355) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$lambda$11(ThreadLocalTransactionManager.kt:402) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:401) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$lambda$5(ThreadLocalTransactionManager.kt:310) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:259) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:237) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:236) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.tabComplete(KickCommandProcessor.kt:126) ~[ImprovedFactions-2.3.1.jar:?]
        at ImprovedFactions-2.3.1.jar/io.github.toberocat.improvedfactions.commands.executor.CommandExecutor.onTabComplete(CommandExecutor.kt:103) ~[ImprovedFactions-2.3.1.jar:?]
        at org.bukkit.command.PluginCommand.tabComplete(PluginCommand.java:138) ~[paper-api-1.21.8-R0.1-SNAPSHOT.jar:?]
        ... 21 more
[23:45:21 INFO]: n0ttws issued server command: /f kick WardenSky
[23:45:21 WARN]: [ImprovedFactions] The command kick used a non localized exception - This is the outdated way of handling exceptions. Exception Class: java.lang.NullPointerException. In the future, please use LocalizedException. In the meantime, this exception will be printed to the console.
[23:45:21 WARN]: java.lang.NullPointerException
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.processCall(KickCommandProcessor.kt:59)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.access$processCall(KickCommandProcessor.kt:21)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor$execute$$inlined$loggedTransaction$1.invoke(DatabaseManager.kt:30)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor$execute$$inlined$loggedTransaction$1.invoke(DatabaseManager.kt:23)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:355)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$lambda$11(ThreadLocalTransactionManager.kt:402)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:401)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$lambda$5(ThreadLocalTransactionManager.kt:310)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:410)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:259)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:237)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:236)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.member.KickCommandProcessor.execute(KickCommandProcessor.kt:119)
[23:45:21 WARN]:        at ImprovedFactions-2.3.1.jar//io.github.toberocat.improvedfactions.commands.executor.CommandExecutor.onCommand(CommandExecutor.kt:133)
[23:45:21 WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[23:45:21 WARN]:        at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:83)
[23:45:21 WARN]:        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
[23:45:21 WARN]:        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:30)
[23:45:21 WARN]:        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:13)
[23:45:21 WARN]:        at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8)
[23:45:21 WARN]:        at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5)
[23:45:21 WARN]:        at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:105)
[23:45:21 WARN]:        at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:451)
[23:45:21 WARN]:        at net.minecraft.commands.Commands.performCommand(Commands.java:357)
[23:45:21 WARN]:        at net.minecraft.commands.Commands.performCommand(Commands.java:345)
[23:45:21 WARN]:        at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2313)
[23:45:21 WARN]:        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$12(ServerGamePacketListenerImpl.java:2286)
[23:45:21 WARN]:        at net.minecraft.server.TickTask.run(TickTask.java:18)
[23:45:21 WARN]:        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:155)
[23:45:21 WARN]:        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1450)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:176)
[23:45:21 WARN]:        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:129)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1430)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1424)
[23:45:21 WARN]:        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:139)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:1381)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1389)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1266)
[23:45:21 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:310)
[23:45:21 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)

### Plugin version

2.3.1

### Server version

1.21.8

--- a/improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/commands/arguments/faction/FactionUserArgumentParser.kt
+++ b/improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/commands/arguments/faction/FactionUserArgumentParser.kt
@@ -1,0 +1,35 @@
+package io.github.toberocat.improvedfactions.commands.arguments.faction
+
+import io.github.toberocat.improvedfactions.annotations.localization.Localization
+import io.github.toberocat.improvedfactions.commands.arguments.ArgumentParser
+import io.github.toberocat.improvedfactions.commands.arguments.ArgumentParsingException
+import io.github.toberocat.improvedfactions.commands.arguments.ParsingContext
+import io.github.toberocat.improvedfactions.user.FactionUser
+import io.github.toberocat.improvedfactions.user.factionUser
+import org.bukkit.command.CommandSender
+import org.bukkit.entity.Player
+
+@Localization("base.arguments.faction-user.not-player")
+@Localization("base.arguments.faction-user.not-found")
+class FactionUserArgumentParser : ArgumentParser {
+    override fun parse(sender: CommandSender, arg: String, args: Array<String>): Any {
+        val user = (sender as? Player)?.factionUser()
+            ?: throw ArgumentParsingException("base.arguments.faction-user.not-player")
+        val faction = user.faction()
+            ?: throw ArgumentParsingException("base.arguments.faction-user.not-found")
+
+        return faction.members().find {
+            it.offlinePlayer().name.equals(arg, ignoreCase = true)
+        } ?: throw ArgumentParsingException("base.arguments.faction-user.not-found")
+    }
+
+    override fun rawTabComplete(pCtx: ParsingContext): List<String> {
+        val user = pCtx.player()?.factionUser() ?: return emptyList()
+        val faction = user.faction() ?: return emptyList()
+
+        return faction.members()
+            .filter { it.uniqueId != user.uniqueId }
+            .mapNotNull { it.offlinePlayer().name }
+            .filter { it.startsWith(pCtx.arg, ignoreCase = true) }
+    }
+}

--- a/improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/commands/executor/CommandExecutor.kt
+++ b/improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/commands/executor/CommandExecutor.kt
@@ -10,6 +10,7 @@ import io.github.toberocat.improvedfactions.commands.arguments.bukkit.WorldArgum
 import io.github.toberocat.improvedfactions.commands.arguments.faction.FactionArgumentParser
 import io.github.toberocat.improvedfactions.commands.arguments.faction.FactionInviteArgumentParser
 import io.github.toberocat.improvedfactions.commands.arguments.faction.FactionRankArgumentParser
+import io.github.toberocat.improvedfactions.commands.arguments.faction.FactionUserArgumentParser
 import io.github.toberocat.improvedfactions.commands.arguments.faction.ZoneArgumentParser
 import io.github.toberocat.improvedfactions.commands.arguments.primitives.BoolArgumentParser
 import io.github.toberocat.improvedfactions.commands.arguments.primitives.IntArgumentParser
@@ -24,6 +25,7 @@ import io.github.toberocat.improvedfactions.modules.base.BaseModule
 import io.github.toberocat.improvedfactions.modules.power.PowerType
 import io.github.toberocat.improvedfactions.ranks.FactionRank
 import io.github.toberocat.improvedfactions.translation.LocalizedException
+import io.github.toberocat.improvedfactions.user.FactionUser
 import io.github.toberocat.improvedfactions.translation.sendLocalized
 import io.github.toberocat.improvedfactions.zone.Zone
 import org.bukkit.OfflinePlayer
@@ -45,7 +47,8 @@ val DEFAULT_PARSERS = mapOf<Class<*>, ArgumentParser>(
     World::class.java to WorldArgumentParser(),
     FactionInvite::class.java to FactionInviteArgumentParser(),
     FactionRank::class.java to FactionRankArgumentParser(),
-    PowerType::class.java to PowerTypeArgumentParser()
+    PowerType::class.java to PowerTypeArgumentParser(),
+    FactionUser::class.java to FactionUserArgumentParser()
 )
 
 open class CommandExecutor(private val plugin: ImprovedFactionsPlugin) : TabExecutor {

